src/megameklab/com/ui/StartupGUI.java

```diff
@@ -0,0 +1,423 @@
+/*
+ * MegaMekLab - Copyright (C) 2019 - The MegaMek Team
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ */
+package megameklab.com.ui;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.DisplayMode;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Image;
+import java.awt.Insets;
+import java.awt.MediaTracker;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.util.ResourceBundle;
+import java.util.TreeMap;
+
+import javax.swing.ImageIcon;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+
+import megamek.client.ui.swing.UnitLoadingDialog;
+import megamek.client.ui.swing.UnitSelectorDialog;
+import megamek.client.ui.swing.widget.MegamekButton;
+import megamek.client.ui.swing.widget.SkinSpecification;
+import megamek.client.ui.swing.widget.SkinXMLHandler;
+import megamek.common.Aero;
+import megamek.common.BattleArmor;
+import megamek.common.Configuration;
+import megamek.common.Entity;
+import megamek.common.EquipmentType;
+import megamek.common.FixedWingSupport;
+import megamek.common.GunEmplacement;
+import megamek.common.Infantry;
+import megamek.common.Mech;
+import megamek.common.Protomech;
+import megamek.common.Tank;
+import megamek.common.util.EncodeControl;
+import megamek.common.util.ImageUtil;
+import megamek.common.util.MegaMekFile;
+import megameklab.com.MegaMekLab;
+import megameklab.com.ui.dialog.LoadingDialog;
+import megameklab.com.util.UnitUtil;
+
+/**
+ * A startup splash screen for MegaMekLab
+ * @author Taharqa
+ */
+public class StartupGUI extends javax.swing.JPanel {
+
+    
+    private static final long serialVersionUID = 8376874926997734492L;
+    JFrame frame;
+    Image imgSplash;
+    BufferedImage backgroundIcon;
+    
+    private MegamekButton btnLoadUnit;
```

Comment:
The button fields aren't ever used outside initComponents() and can be converted to local variables.

Reply:
fixed in the latest